osu.Server.Queues.ScorePump/ImportHighScores.cs

```diff
@@ -76,32 +66,23 @@ public int OnExecute(CancellationToken cancellationToken)
                             ruleset_id = RulesetId,
                             passed = true,
                             total_score = highScore.score,
-                            accuracy = scoreInfo.Accuracy,
+                            accuracy = accuracy,
                             max_combo = highScore.maxcombo,
                             rank = Enum.TryParse(highScore.rank, out ScoreRank parsed) ? parsed : ScoreRank.D,
                             mods = ruleset.ConvertFromLegacyMods((LegacyMods)highScore.enabled_mods).Select(m => new APIMod(m)).ToList(),
-                            statistics = scoreInfo.Statistics,
-                            started_at = highScore.date,
-                            ended_at = highScore.date,
-                            created_at = highScore.date,
-                            updated_at = highScore.date
+                            statistics = statistics,
                         },
                         created_at = highScore.date,
                         updated_at = highScore.date,
                     };
 
-                    // Todo: Import highScore.hidden somehow?
-
                     using (var transaction = db.BeginTransaction())
                     {
-                        soloScore.id = db.Insert(soloScore, transaction);
+                        long insertId = db.Insert(soloScore, transaction);
 
-                        // Update data to match the row ID.
-                        db.Execute($"UPDATE {SoloScore.TABLE_NAME} s SET s.data = JSON_SET(s.data, '$.id', @id) WHERE s.id = @id", soloScore, transaction);
```

Comment:
osu!web _does_ do this: https://github.com/ppy/osu-web/blob/6051b8855002ca1e339acc34dfdb984cfea74483/app/Http/Controllers/Solo/ScoresController.php#L52-L54